contract/contracts/Treasury/EVMTreasury.sol

```diff
@@ -3,102 +3,147 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
-// import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
-import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../Library/Verify.sol";
+import "../Library/BytesLib.sol";
+import "../Library/Utils.sol";
+import "../Library/Strings.sol";
 import "./interfaces/IEVMTreasury.sol";
 
-contract EVMTreasury is Pausable, ReentrancyGuard, IEVMTreasury {
+contract EVMTreasury is ReentrancyGuard, IERC721Receiver, IEVMTreasury {
+    using BytesLib for bytes;
+
     /// @notice The name of this contract
     string public constant name = "EVM SETTLEMENT CHAIN TREASURY V1";
+    bytes public constant chainName = hex"6d797468657265756d"; // mythereum, for testing
+    uint128 public constant contractSequence = 0;
 
-    Client public client;
-
-    mapping(uint256 => Client) public clients;
+    LightClient public lightClient;
 
+    mapping(uint256 => LightClient) public lightClients;
```

Comment:
Add public variables to the interface.
i.e. `function lightClient() returns (LightClient)`

Reply:
Actually, we need a separate view function for commitRoots. I will add both.

Reply:
Added.

Reply:
I think `contractSequence` could also be the query target. (and maybe `name` and `chainName`)

Reply:
All targets(public variables) are added.